app/javascript/controllers/search_controller.js

```diff
@@ -10,13 +10,31 @@ export default class extends Controller {
   submit() {
     clearTimeout(this.timeout)
     this.timeout = setTimeout(() => {
-      this.submitForm()
+      const next = this.encodeForm()
+      if (next !== window.location.href) {
+        this.submitForm()
+        this.setHistory(next)
```

Comment:
i think in the midst of learning how turbo frames worked, we assumed that this might have something to do with an input refresh issue we were running into, but couldn't explain. we can probably leave this out of the PR if it doesn't make sense here.

Reply:
So normally, turbo-frames do _not_ change your browser history/url.  They only update the interior of that turbo-frame tag on the page.

I think when i wrote this controller for Augury, I wanted the history to change.  So I cobbled together some stuff.  That may have been a bad choice - it feels very javascript-heavy.

I need to do more reading/trying-things, but turbo _streams_ might be a better solution.